components/org.apache.stratos.autoscaler/src/main/java/org/apache/stratos/autoscaler/client/AutoscalerCloudControllerClient.java

```diff
@@ -48,194 +48,205 @@
-            log.error(msg, e);
-        }
-    }
-
-
-    private Volume[] convertVolumesToStubVolumes(VolumeContext[] volumeContexts) {
-
-        ArrayList<Volume> volumes = new ArrayList<Volume>();
-        for (VolumeContext volumeContext : volumeContexts) {
-            Volume volume = new Volume();
-            volume.setRemoveOntermination(volumeContext.isRemoveOntermination());
-            volume.setMappingPath(volumeContext.getMappingPath());
-            volume.setId(volumeContext.getId());
-            volume.setDevice(volumeContext.getDevice());
-            volume.setIaasType(volumeContext.getIaasType());
-            volume.setSnapshotId(volumeContext.getSnapshotId());
-            volume.setVolumeId(volumeContext.getVolumeId());
-            volume.setSize(volumeContext.getSize());
-            volumes.add(volume);
-        }
-        return volumes.toArray(new Volume[volumes.size()]);
-    }
-
-    public void terminateInstance(String memberId) throws Exception {
-        if (log.isInfoEnabled()) {
-            log.info(String.format("Terminating instance via cloud controller: [member] %s", memberId));
-        }
-        long startTime = System.currentTimeMillis();
-        stub.terminateInstance(memberId);
-        if (log.isDebugEnabled()) {
-            long endTime = System.currentTimeMillis();
-            log.debug(String.format("Service call terminateInstance() returned in %dms", (endTime - startTime)));
-        }
-    }
-
-    public void terminateInstanceForcefully(String memberId) throws Exception {
-        if (log.isDebugEnabled()) {
-            log.debug(String.format("Terminating instance forcefully via cloud controller: [member] %s", memberId));
-        }
-        stub.terminateInstanceForcefully(memberId);
-    }
-
-    public void terminateAllInstances(String clusterId) throws RemoteException,
-            CloudControllerServiceInvalidClusterExceptionException {
-        if (log.isInfoEnabled()) {
-            log.info(String.format("Terminating all instances of cluster via cloud controller: [cluster] %s", clusterId));
-        }
-        long startTime = System.currentTimeMillis();
-        stub.terminateInstances(clusterId);
-
-        if (log.isDebugEnabled()) {
-            long endTime = System.currentTimeMillis();
-            log.debug(String.format("Service call terminateInstances() returned in %dms", (endTime - startTime)));
-        }
-    }
+	private static final Log log = LogFactory.getLog(AutoscalerCloudControllerClient.class);
+
+	private static CloudControllerServiceStub stub;
+
+	/* An instance of a CloudControllerClient is created when the class is loaded.
+	 * Since the class is loaded only once, it is guaranteed that an object of
+	 * CloudControllerClient is created only once. Hence it is singleton.
+	 */
+	private static class InstanceHolder {
+		private static final AutoscalerCloudControllerClient INSTANCE = new AutoscalerCloudControllerClient();
+	}
```

Comment:
I know this code is there from the beginning. But why do we need this inner class? IMO, we don't need an inner class to make a parent class a singleton.